src/resonate/record.py

```diff
@@ -54,7 +58,21 @@ def __init__(
         self._f = Future[T]()
         self.children: list[Record[Any]] = []
         self.invocation: LFI | RFI = invocation
-        self.retry_policy = invocation.opts.retry_policy
+        self.retry_policy: retry_policy.RetryPolicy | None
+        if not isinstance(invocation.unit, Invocation):
+            self.retry_policy = None
+        elif isinstance(invocation.unit.fn, str):
+            self.retry_policy = invocation.opts.retry_policy or None
+        elif isgeneratorfunction(invocation.unit.fn):
+            self.retry_policy = invocation.opts.retry_policy or never()
+        else:
+            assert iscoroutinefunction(invocation.unit.fn) or isfunction(
+                invocation.unit.fn
+            )
+            self.retry_policy = invocation.opts.retry_policy or constant(
```

Comment:
what about exponential as the default?

Reply:
I'll then set a maximum exponential backoff time in the option. Backoff up to 30secs, after that will be constant 30 secs?

How do you feel about it?